core/src/main/scala/org/apache/spark/TestUtils.scala

```diff
@@ -173,21 +173,23 @@ private[spark] object TestUtils {
    * Run some code involving jobs submitted to the given context and assert that the jobs spilled.
    */
   def assertSpilled[T](sc: SparkContext, identifier: String)(body: => T): Unit = {
-    val spillListener = new SpillListener
-    sc.addSparkListener(spillListener)
-    body
-    assert(spillListener.numSpilledStages > 0, s"expected $identifier to spill, but did not")
+    withListener(sc, new SpillListener) { listener =>
+      val ret = body
+      assert(listener.numSpilledStages > 0, s"expected $identifier to spill, but did not")
```

Comment:
Maybe I'm missing something obvious, but shouldn't we run the `body` before assert?